nilearn/decomposition/canica.py

```diff
@@ -106,52 +107,42 @@ class CanICA(MultiPCA, CacheMixin):
       datasets", IEEE ISBI 2010, p. 1177
     """
 
-    def __init__(self, mask=None, n_components=20,
-                 smoothing_fwhm=6, do_cca=True,
-                 threshold='auto', n_init=10,
-                 standardize=True,
-                 random_state=0,
-                 target_affine=None, target_shape=None,
+    def __init__(self, n_init=10, threshold='auto', n_components=20,
+                 do_cca=True,
+                 random_state=None,
+                 mask=None, smoothing_fwhm=None,
+                 standardize=True, detrend=True,
                  low_pass=None, high_pass=None, t_r=None,
-                 # Common options
+                 target_affine=None, target_shape=None,
+                 mask_strategy='epi', mask_args=None,
                  memory=Memory(cachedir=None), memory_level=0,
-                 n_jobs=1, verbose=0,
+                 n_jobs=1, verbose=0
                  ):
+
         super(CanICA, self).__init__(
-            mask=mask, memory=memory, memory_level=memory_level,
-            n_jobs=n_jobs, verbose=verbose, do_cca=do_cca,
-            n_components=n_components, smoothing_fwhm=smoothing_fwhm,
-            target_affine=target_affine, target_shape=target_shape)
+            n_components=n_components,
+            do_cca=do_cca,
+            random_state=random_state,
+            # feature_compression=feature_compression,
+            mask=mask, smoothing_fwhm=smoothing_fwhm,
+            standardize=standardize, detrend=detrend,
+            low_pass=low_pass, high_pass=high_pass, t_r=t_r,
+            target_affine=target_affine, target_shape=target_shape,
+            mask_strategy=mask_strategy, mask_args=mask_args,
+            memory=memory, memory_level=memory_level,
+            n_jobs=n_jobs, verbose=verbose)
         self.threshold = threshold
-        self.random_state = random_state
-        self.low_pass = low_pass
-        self.high_pass = high_pass
-        self.t_r = t_r
         self.n_init = n_init
-        self.standardize = standardize
 
-    def fit(self, imgs, y=None, confounds=None):
-        """Compute the mask and the ICA maps across subjects
-
-        Parameters
-        ----------
-        imgs: list of Niimg-like objects
-            See http://nilearn.github.io/building_blocks/manipulating_mr_images.html#niimg.
-            Data on which PCA must be calculated. If this is a list,
-            the affine is considered the same for all.
-
-        confounds: CSV file path or 2D matrix
-            This parameter is passed to nilearn.signal.clean. Please see the
-            related documentation for details
-        """
-        MultiPCA.fit(self, imgs, y=y, confounds=confounds)
+    def _unmix_components(self):
```

Comment:
Can you do a short one-line docstring here.

The reason being that it helps coders (not end users) understand the purpose of the function.